### DISCUSSION TITLE
Graphs resolution for past days

### DISCUSSION THREAD

**Discussion by sfkristof** [javamelody/javamelody#1139] [Q&A] [answered]
Hi!

First of all: a great and very useful tool, big thanks!  👍 

My questions are about the graphs:
1) Why is the resolution so coarse and 1 hour based when selecting a customized (but one day long) date period?
If I check the current day then the graphs are good and detailed (1 minute based I guess):
![Screenshot_35](https://user-images.githubusercontent.com/22003889/213793210-192d23d7-339c-4752-bdf1-d03ce690a779.png)
But when I check yesterday, it's very coarse:
![Screenshot_36](https://user-images.githubusercontent.com/22003889/213793287-118220ae-1faa-4b5c-a17e-dbae23b6bd0a.png)
Is it a config error?
I have v1.79.0 but I didn't see any feature covering this topic in the release notesup to the newst version.

2) Would it be possible to somehow filter for a specific time period as well?
It's often the case that we should check for a 5-10 minute problematic time range and check the http requests, sql queries, number of transactions, etc.
Could the "Customized" date filter be extended to minutes maybe?

Thanks in advance!

BR,
  Kristof

**Comment by mmoser18** [comment]
I would second this request/question! Would it be possible to keep the full resolution for, say, a week and only start compressing details for older periods? We are seeking to identify certain "patterns" in our application's behavior and for that it would be very helpful, if one could compare and display several days in full resolution. Of course one can always work with screen shots but that's a bit tedious, as one always has to take these at more or less the same time to obtain comparable results.

**Comment by evernat** [comment, marked as answer]
1. Data of graphs are stored in Round Robin Databases (rrd files). See https://en.wikipedia.org/wiki/RRDtool And so, one value is stored every minute for one day, then values are consolidated with one value stored every hour for one week, same for one month and for 2 years : size of a rrd file is 34 kb from creation to forever, even 10 years after. Values are consolidated like that so that total storage does not take gigabytes but more like 10 or 100 megabytes. So you can't have values every minute for whatever period you choose, except the current day.
2. Data of statistics of http requests and sql queries are aggregated, per day, for the same reason that the storage must not take gigabytes. So you can't have statistics for a 5 minutes period. (But in the "All" period, you can click to clear the statistics without impacting the "Day" period, so you can have the http and sql statistics for one or some minutes after that. It is useful for a few tests in QA.)

**Reply by sfkristof** [reply]
Ok, but if I have HDD I can't choose to keep the data fow a week or so to go back after a few days to check something?
Melody is only good for current day or for "high level" trends/statistics..?

**Reply by pier4r** [reply]
@evernat totally understandable that the RRD tool wants to be efficient, but can one tell it "please for the last months keep the default resolution, do not consolidate the data, we have enough space" ? Because if one is willing to commit the space for the usefulness of the data, then why not?

I am trying to find the configuration to tell RRD to avoid consolidation too early but I am finding none.

E: I see, in this page: https://github.com/javamelody/javamelody/wiki/UserGuideAdvanced#sending-metrics-to-influxdb the suggestion is to send stuff to other databases for custom management.

**Reply by evernat** [reply]
@pier4r Sorry, it is not possible to change the resolution of the week period or of the month period only. But you can change a bit the resolution of all the periods (day, week, month, year) if you want, with the `resolution-seconds` javamelody parameter as said in the [user's guide](https://github.com/javamelody/javamelody/wiki/UserGuide#6-optional-parameters). A resolution of 30 seconds instead of the default 60 seconds would increase a bit the negligeable overhead of javamelody and would increase a bit the size of rrd files in the storage directory. If you change the `resolution-seconds` parameter, you need to delete the rrd files in the storage directory in order to be effective and so you will lose the values in graphs which were in the rrd files before you delete them.

Sending metrics to other databases is not a suggestion at all. This doc just says that javamelody is not closed about sending metrics to influxdb or prometheus or others if you want to and it also says that javamelody already has integrations for that.
